### PR TITLE
Udpate the "survey update" functionality

### LINKED ISSUE #129: post API update according to the new survey settings UI design
When "action=add_or_update" and "what=surveys", post API needs to decide whether to clone a new survey, update the survey then or purely update the survey.

### PR DESCRIPTION
**Pull request by DarrenZhang01**
(If `action=add_or_update`, the API will check whether the survey has
already been set before; If not, a new survey will be branched first and
then update on the newly branched survey)

fixes #129

### CODE REVIEW

File: post_info.php [hunk 1]
```diff
@@ -320,6 +320,25 @@ function handle_survey_setting($survey_id, $level, $user_id, $action, $data)
     $section_id = null;
     // If the user wants to update a survey
     if ($action == "add_or_update") {
+        // Use preprocessing to decide what to do, to be more precise:
+        //     a. If the survey_choice at the level of the user is null,
+        //        we branch first and then update
+        //     b. If the survey_choice at the level of the user is not null,
+        //        we directly update this survey
+        $decision = determine_action_on_add_or_update($survey_id, $level);
+        if ($decision == "branch_and_update") {
```
Comment (siefkenj) [inline post_info.php]: A `switch` statement would be better here.

File: post_info.php [hunk 2]
```diff
@@ -347,6 +367,34 @@ function handle_survey_setting($survey_id, $level, $user_id, $action, $data)
     }
 }
 
+function determine_action_on_add_or_update($survey_id, $level)
+{
+    $survey_choice = null;
+    $bind_variables = array("survey_id" => $survey_id);
+    $sql = null;
+    if ($level == "dept") {
```
Comment (siefkenj) [inline post_info.php]: `switch` statement.

Comment (DarrenZhang01) [reply post_info.php]: Sure!

File: post_info.php (hunk 1 above)
Comment (DarrenZhang01) [reply post_info.php]: Sure!

**Pull request merged**